crates/holochain/src/core/ribosome/real_ribosome.rs

```diff
@@ -208,12 +216,97 @@ fn context_key_from_key(key: &[u8; 32]) -> u64 {
 
 impl RealRibosome {
     /// Create a new instance
-    pub fn new(dna_file: DnaFile) -> Self {
-        Self { dna_file }
+    pub fn new(dna_file: DnaFile) -> RibosomeResult<Self> {
+        // Create an empty ribosome.
+        let ribosome = Self {
+            dna_file,
+            zome_types: Default::default(),
+            zome_dependencies: Default::default(),
+        };
+
+        // Collect the number of entry and link types
+        // for each integrity zome.
+        let iter = ribosome
+            .dna_def()
+            .integrity_zomes
+            .iter()
+            .map(|(name, zome)| {
+                let zome = Zome::new(name.clone(), zome.clone().erase_type());
+
+                // Call the const functions that return the number of types.
+                let num_entry_types = match ribosome.get_const_fn(&zome, "__num_entry_types")? {
```

Comment:
It could be problematic to use `u8` instead of something larger to represent the count of entry types, since we can't express having all 256 entry types. Unless there is an explicit limit on having no more than 255, with index `255` being reserved or something.

Same for link types obviously.